.github/workflows/deploy-to-production.yml

```diff
@@ -0,0 +1,219 @@
+name: Deploy to Production
+
+# Schedule deployment to Production
+on:
+  repository_dispatch:
+    types:
+      - slack-prod-deploy-command-odv2
+  schedule:
+    # Runs at 12:05 pm weekdays (UK time)
+    - cron: "5 12 * * 1-5"
+
+
+env:
+  #To be replaced by Prod Env Id 
+  PROD_ENV_ID: "083ff91e-6ac5-4638-945c-3ee18acc40bf"
+  TARGET: "prod"
+jobs:
+
+  integrity:
+    name: Slack Integrity Check
+    # Only runs for SLACK Command (hotfix)
+    if: ((github.event_name != 'schedule'))
+    runs-on: ubuntu-latest
+    timeout-minutes: 1
+    steps:
+      # Validate that the command is called from deployment-od2-prod
+      - name: Checking source
+        uses: actions/github-script@v3
+        if: ${{ env.CHANNEL_ID !='C0353KHFXNJ' }}
+        with:
+          script: |
+            core.setFailed('Caller source not verified!')
+
+      #
+      # Validate that the origin is from Slack
+      # See: https://api.slack.com/authentication/verifying-requests-from-slack
+      #
+      - name: Validate Origin
+        uses: actions/github-script@v4.0.2
+        env:
+          AGENT: ${{ github.event.client_payload.integrity.agent }}
+          BODY: ${{ github.event.client_payload.integrity.body }}
+          TIMESTAMP: ${{ github.event.client_payload.integrity.timestamp }}
+          SIGNATURE: ${{ github.event.client_payload.integrity.signature }}
+          SIGNING_SECRET: ${{ secrets.SLACK_SIGNING_SECRET }}
+        with:
+          script: |
+            if(!`${process.env.AGENT}`.startsWith("Slackbot")){
+                core.setFailed("User agent mismatch.")
+            } else {
+                const base = `v0:${process.env.TIMESTAMP}:${process.env.BODY}`
+                const hash = require('crypto').createHmac('sha256', `${process.env.SIGNING_SECRET}`).update(base).digest("hex")
+                if(`${process.env.SIGNATURE}` !== `v0=${hash}`) {
+                  core.setFailed("The origin integrity check failed.")
+                }
+            }
+
+      #
+      # Provide a URL to the active GitHub Actions Workflow run
+      #
+      - name: Slack message (Track progress message)
+        if: ${{ success() }}
+        uses: muinmomin/webhook-action@v1.0.0
+        with:
+          url: ${{ env.SLACK_WEBHOOK }}
+          data: '{ "channel": "${{ env.CHANNEL_ID }}", "text": "Pipeline progress...", "attachments": [{ "text" : "The pipeline logs are here ${{ env.GITHUB_WORKFLOW_URL }}",  "color": "#005EB8" }] }'
+        env:
+          SLACK_WEBHOOK: ${{  secrets.SLACK_WEBHOOK }}
+          GITHUB_WORKFLOW_URL: "https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}"
+          CHANNEL_ID: ${{ github.event.client_payload.data.channel_id }}
+
+
+  prepare-prod-deployment:
+    name: Deploy to Production
+    runs-on: ubuntu-latest
+    timeout-minutes: 85
+    steps:
+      - if: ((github.event_name != 'schedule')&&(github.event_name != 'push'))
```

Comment:
A few notes on this tag move. 

- Even though there is a cost to spinning up a separate job, I think it would feel cleaner to move this tag shift to its own job. 
- Wouldn't you need to have checked out the repository before moving tags?
- There is no trigger for 'push'. I am wondering why you have checked for '!= push'?